pkg/tasks/certs.go

```diff
@@ -31,3 +42,100 @@ func deployCAOnNode(s *state.State, node *kubeoneapi.HostConfig, conn ssh.Connec
 	s.Logger.Infoln("Uploading PKI files...")
 	return s.Configuration.UploadTo(conn, s.WorkDir)
 }
+
+func renewCerts(s *state.State) error {
+	if !s.ForceUpgrade {
+		s.Logger.Warn("Your control-plane certificates are about to expire in less then 90 days")
+		s.Logger.Warn("To renew them without changing kubernetes version run `kubeone apply --force-upgrade`")
+		return nil
+	}
+	s.Logger.Warn("Your control-plane certificates are about to expire in less then 90 days")
+	s.Logger.Warn("Force renewing Kubernetes certificates")
+
+	// /etc/kubernetes/admin.conf will be changed after certificates renew, so we have to initialize client again
+	s.Logger.Infoln("Resetting Kubernetes clientset...")
+	s.DynamicClient = nil
+
+	renewCmd := "sudo kubeadm alpha certs renew all"
+	greaterThen120, _ := semver.NewConstraint(">=1.20")
+	if greaterThen120.Check(s.LiveCluster.ExpectedVersion) {
+		renewCmd = "sudo kubeadm certs renew all"
+	}
+
+	err := s.RunTaskOnControlPlane(
+		func(s *state.State, node *kubeoneapi.HostConfig, conn ssh.Connection) error {
+			_, _, _, err := conn.Exec(renewCmd)
+			return err
+		},
+		state.RunParallel,
+	)
```

Comment:
What about certificates on the static worker nodes? Do we maybe need to renew those as well?

Reply:
There are no certificates to renew with kubeadm on worker nodes.
At the moment we have kubelet server self-signed cert (which is already "not-valid"). Plus kubelet-client cert that is rotated by kubelet itself. Everything else works based on tokens.